doc/source/user_manual/configuration/connector.rst

```diff
@@ -103,10 +140,12 @@ Example
         auto_commit: on
         session_timeout: 6000
         offset_reset_policy: smallest
-        hmac:
-          target: <RAW_MSG>
-          key: secret-key
-          output_field: Hmac
+        preprocessing:
```

Comment:
The consumer is specific to the confluent-kafka connector. Preprocessors will be useful for other inputs as well. Do you think placing it under 'consumer' is the best choice?

Reply:
I completely agree with you that's why I already implemented appending the version info in the pipeline.py instead of the confluent_kafka.py, which currently has the hmac functionality implemented. I just left the hmac calculation where it is for now as we were already talking about doing a refactoring of the inputs and outputs in general. If I would have already moved the hmac functionality away from the confluent_kafka then this merge request would have been much larger. 

I'd say we leave it as is for now an, discuss what the best solution for inputs, outputs and preprocessing is and then start working on how to implement it. Especially if we plan to add more preprocessors we should find a common interface for them before they get too messy.